src/App.tsx

```diff
@@ -61,14 +82,24 @@ export const App: React.FC = () => {
       filteredMarkets.map(currentMarket => {
         return getData(`mktDetail?id=${currentMarket.id}`)
           .then(response => checkForError(response))
-          .then(response => response.json());
+          .then(response => response.json())
+          .then(data => cleanDetailsData(data.marketdetails, currentMarket.id))
       })
     )
-      .then(arrayOfPromises => cleanDetailsData(arrayOfPromises))
-      .then(cleanData => setDetails(cleanData))
-      .catch(error => setError(error));
+    .then(data => addScheduleToMarkets(data, filteredMarkets))
+    .then(completeData => setData(completeData))
   };
 
+  const setData = (data: ApiMarkets) => {
+    setMarkets(data.markets)
+    setDetails(data.marketDetails)
+  }
+
+  const findSelectedMarket = (marketID: number) => {
+    setSelectedMarket(marketDetails.find(market => market.id === marketID))
```

Comment:
Personal preference, but I would prefer to see the logic for the find set to a variable and then set that variable in state.